packages/tcore-sdk/src/Types/Plugin/Plugin.types.test.ts

```diff
@@ -0,0 +1,214 @@
+    expect(parsed.domain).toBeUndefined;
+  });
+});
+
+describe("zPluginInstallOptions", () => {
+  test("accepts empty object", () => {
+    const data = {};
+    const parsed = zPluginInstallOptions.parse(data);
+    expect(parsed.start).toBeUndefined;
+    expect(parsed.log).toBeUndefined;
+    expect(parsed.restoreBackup).toBeUndefined;
+  });
+});
+
+describe("zPluginPermission", () => {
+  test("invalid option", () => {
+    const data = "a";
+    try {
+      zPluginPermission.parse(data);
+    } catch (error) {
+      const e = (error as ZodError).flatten();
+      expect(e.formErrors[0]).toBe(
+        "Invalid enum value. Expected 'device' | 'analysis' | 'action' | 'device-data' | 'cli'"
+      );
+    }
+  });
+});
+
+describe("zPluginManifestCliOption", () => {
+  test("check optional field", () => {
+    const data = {
+      flags: "flags",
+    };
+    const parsed = zPluginManifestCliOption.parse(data);
+    expect(parsed.description).toBeUndefined;
+  });
+});
+
+describe("zPluginManifestCliCommand", () => {
+  test("error if malformed array", () => {
+    const data = {
+      name: "name",
+      arguments: [{ flags: 0 }],
+      file: "file",
+    };
+    try {
+      zPluginManifestCliCommand.parse(data);
+    } catch (error) {
+      const e = (error as ZodError).flatten();
+      expect(e.fieldErrors).toStrictEqual({ arguments: ["Expected string, received number"] });
+    }
+  });
+});
+
+describe("zPluginPackageTCore", () => {
+  test("error if malformed array", () => {
+    const data = {
+      name: "name",
+      types: ["a"],
+    };
+    try {
+      zPluginPackageTCore.parse(data);
+    } catch (error) {
+      const e = (error as ZodError).flatten();
+      expect(e.fieldErrors.types).toStrictEqual([
+        "Invalid enum value. Expected 'action-trigger' | 'action-type' | 'database' | 'decoder' | 'encoder' | 'filesystem' | 'hook' | 'navbar-button' | 'page' | 'service' | 'sidebar-button' | 'sidebar-footer-button'",
+      ]);
+    }
+  });
+});
+
+describe("zPluginConfigField", () => {
+  test("invalid type", () => {
+    const data = {};
+    try {
+      zPluginConfigField.parse(data);
+    } catch (error) {
+      const e = (error as ZodError).flatten();
+      expect(e.formErrors[0]).toBe("Invalid input");
+    }
+  });
+});
+
+describe("zPluginStorageItemSet", () => {
+  test("assure correct type", () => {
+    const data = {
+      value: 0,
+      key: "key",
+    };
+    const parsed = zPluginStorageItemSet.parse(data);
+    expect(parsed.type).toEqual("number");
+  });
+});
+
+describe("zPluginModule", () => {
+  test("invalid option", () => {
+    const data = {
+      state: "state",
+      message: 0,
+    };
+    try {
+      zPluginModule.parse(data);
+    } catch (error) {
+      const e = (error as ZodError).flatten();
+      expect(e.fieldErrors.state).toStrictEqual([
+        "Invalid enum value. Expected 'idle' | 'stopping' | 'starting' | 'started' | 'stopped'",
```

Comment:
Don't add all the values in test, it's going to be difficult to change it later on